src/components/tables/Table.tsx

```diff
@@ -161,12 +159,47 @@ export class Table extends React.Component<ITableProps, {}> {
         <table className={tableClasses}>
           <TableChildLoadingRow {...this.props} isInitialLoad={this.isInitialLoad} />
           <TableChildHeader {...this.props} />
-          {TableChildBody(this.props)}
+          {this.getTableBody()}
         </table>
         <TableChildBlankSlate {...this.props} isInitialLoad={this.isInitialLoad} />
         <TableChildNavigation {...this.props} />
         <TableChildLastUpdated {...this.props} />
       </div>
     );
   }
+
+  private hasTableCompositeStateChanged(currentTableCompositeState: ITableCompositeState, nextTableCompositeState: ITableCompositeState): boolean {
+    return !!currentTableCompositeState && (
+      currentTableCompositeState.filter !== nextTableCompositeState.filter
+      || currentTableCompositeState.perPage !== nextTableCompositeState.perPage
+      || currentTableCompositeState.page !== nextTableCompositeState.page
+      || currentTableCompositeState.sortState.attribute !== nextTableCompositeState.sortState.attribute
+      || currentTableCompositeState.sortState.order !== nextTableCompositeState.sortState.order
+      || (_.isEmpty(currentTableCompositeState.predicates) && !_.isEmpty(nextTableCompositeState.predicates))
+      || _.some(
+        currentTableCompositeState.predicates,
+        (attributeValue: any, attributeName: string) => attributeValue !== nextTableCompositeState.predicates[attributeName],
+      )
+    );
+  }
+
+  private getTableBody() {
+    const tableData = this.props.tableCompositeState.data || this.props.initialTableData;
+
+    return tableData.displayedIds.map((id: string, yPosition: number): JSX.Element => {
+      const rowData: IData = tableData.byId[id];
+
+      return (<TableChildBody
+        key={id}
+        tableId={this.props.id}
+        rowData={rowData}
+        yPosition={yPosition}
```

Comment:
Why do we need yPosition? Can't we use rowData.id for the key? (It was probably like this before). It seems to be use to generate the key of the row https://github.com/coveo/react-vapor/pull/465/files#diff-cfa237717e086f207d2c932bb4d33246R34

Reply:
You're right. Removed and simplified the key.

Reply:
IIRC it's true that it was use for the key of the cells of the row...might not be needed if you use the row id with the x position of the cell or something